examples/tox21/keras_gc.py

```diff
@@ -0,0 +1,253 @@
+"""
+Script that trains multitask models on Tox21 dataset.
+"""
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import numpy as np
+import tensorflow as tf
+
+import deepchem as dc
+from deepchem.molnet import load_tox21
+from feat.mol_graphs import ConvMol
+from tensorflow.python.keras import backend as K
+
+sess = tf.Session()
```

Comment:
more complex models require you to do stuff like K.set_session(session) which sets global state.

Reply:
I think that's because you're using classes from standalone Keras.

Reply:
keras_gc only uses classes from the Tensorflow implementation.  

I had to do K.set_session() and K.set_learning_phase() in order for tf.keras.layers.BatchNormalization to work, or I could of done what TensorGraph does and create a tf.variable for training.

```python
# implementation of BatchNormalization#call
  def call(self, inputs, training=None):
    if training is None:
      training = K.learning_phase()
    output = super(BatchNormalization, self).call(inputs, training=training)
    if training is K.learning_phase():
      output._uses_learning_phase = True  # pylint: disable=protected-access
    return output
```

Reply:
yeah you are right -- we can probably get around this in all cases.